core/cap-0063.md

```diff
@@ -0,0 +1,447 @@
+This is similar to the current meta schema, where the starting value of a LedgerEntry is emitted as
+`LEDGER_ENTRY_STATE` followed by the change. This is similar, except that `LEDGER_ENTRY_STATE` is replaced
+by `LEDGER_ENTRY_RESTORE`.
+
+#### Deprecation of RestoreFootprintOp(?)
+
+With the addition of automatic restoration, there is currently no need for `RestoreFootprintOp`. Functionally, the same automatic
+restoration behavior that exists in `InvokeHostFunctionOp` could be replicated in `ExtendFootprintTTLOp` to provide a single
+"admin" op type for managing lifetimes and restoration.
+
+Currently, `ExtendFootprintTTLOp` is rarely called, as most rent management is performed via contract calls. With the addition of
+automatic restoration, I expect that `RestoreFootprintOp` will also be rarely invoked. It may make sense from a maintenance and UX
+perspective to deprecate `RestoreFootprintOp` in favor of a single op type.
+
+#### getledgerentry Endpoint
+
+In order to facilitate state queries for RPC preflight, captive-core will expose the following HTTP endpoint:
+
+`getledgerentry`
+
+Used to query both live and archived LedgerEntries. While `getledgerentryraw` does simple key-value lookup
+on the live ledger, `getledgerentry` will query a given key in both the live BucketList and Hot Archive BucketList.
+It will also report whether an entry is archived, evicted, or live, and return the entry's current TTL value.
+
+A POST request with the following body:
+
+```http
+ledgerSeq=NUM&key=Base64&key=Base64...
+```
+
+- `ledgerSeq`: An optional parameter, specifying the ledger snapshot to base the query on.
+  If the specified ledger in not available, a 404 error will be returned. If this parameter
+  is not set, the current ledger is used.
+- `key`: A series of Base64 encoded XDR strings specifying the `LedgerKey` to query. TTL keys
+  must not be queried and will return 400 if done so.
+
+A JSON payload is returned as follows:
+
+```json
+{
+"entries": [
+     {"e": "Base64-LedgerEntry", "state": "live", /*optional*/ "ttl": uint32},
+     {"e": "Base64-LedgerKey", "state": "new"},
+     {"e": "Base64-LedgerEntry", "state": "archived"},
+     {"e": "Base64-LedgerEntry", "state": "evicted"}
+],
+"ledger": ledgerSeq
+}
+```
+
+- `entries`: A list of entries for each queried LedgerKey. Every `key` queried is guaranteed to
+have a corresponding entry returned.
+- `e`: Either the `LedgerEntry` or `LedgerKey` for a given key encoded as a Base64 string. If a key
+is live or archived, `e` contains the corresponding `LedgerEntry`. If a key does not exist
+(including expired temporary entries) `e` contains the corresponding `LedgerKey`.
+- `state`: One of the following values:
+  - `live`: Entry is live.
+  - `new`: Entry does not exist. Either the entry has never existed or is an expired temp entry.
+  - `archived`: Entry is archived, but not yet evicted, counts towards in-memory resources.
```

Comment:
Same thought here if we should call expired for entries in live BL and archived for entries that are evicted from live and are in hot archive.

Reply:
This way archived entries count towards disk resources while expired ones are in-memory.